File: src/main/java/com/learnershi/rclasssocket/entity/ClassRoomInfo.java
package com.learnershi.rclasssocket.entity;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.learnershi.rclasssocket.entity.enums.ClassState;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;
import org.springframework.util.StringUtils;

/**
 * ClassRoom
 *
 */
@Accessors(chain = true)
@NoArgsConstructor
@Getter
@Setter
public class ClassRoomInfo {

    String id;                  // 교실 id

    String teacherSeq;          // 강사 seq

    String teacherName;     // 강사 닉네임

    @JsonProperty("roomState")
    ClassState roomState;       // 교실 상태

    String title;               // 교실 제목

    public ClassRoomInfo(ClassRoom classRoom){
        this.id = classRoom.getId();
        this.teacherSeq = classRoom.getTeacherSeq();
        this.teacherName = classRoom.getTeacherName();
        this.roomState = classRoom.getRoomState();
        this.title = StringUtils.hasText(classRoom.getTitle()) ? classRoom.getTitle() : "Study Room";
    }


    @Override
    public boolean equals(Object obj){
        if(obj == null) return false;
        if(obj == this) return true;
        if(!(obj instanceof ClassRoomInfo)) return false;
        ClassRoomInfo classRoomInfo = (ClassRoomInfo) obj;
        return this.id.equals(classRoomInfo.getId());
    }

}

File: src/main/java/com/learnershi/rclasssocket/entity/ClassRoomInfo.kt
package com.learnershi.rclasssocket.entity

import com.fasterxml.jackson.annotation.JsonProperty
import com.learnershi.rclasssocket.entity.enums.*

/**
 * ClassRoom
 *
 */
class ClassRoomInfo(classRoom: ClassRoom) {
    // 교실 id
    var id: String

    // 강사 seq
    var teacherSeq: String?

    // 강사 닉네임
    var teacherName: String?

    // 교실 상태
    @JsonProperty("roomState")
    var roomState: ClassState?

    var title: String

    init {
        id = classRoom.id!!
        teacherSeq = classRoom.teacherSeq
        teacherName = classRoom.teacherName
        roomState = classRoom.roomState
        title = classRoom.title ?: "Study Room"
    }

    override fun equals(obj: Any?): Boolean {
        if (obj == null) return false
        if (obj === this) return true
        if (obj !is ClassRoomInfo) return false
        return id == obj.id
    }
}